### PR TITLE
fix: icon fixes and added new icons

### PR DESCRIPTION
**Pull request by albinAppsmith**
## Description

This PR has added some new icons, added isFullWidth param to segmented control.

Fixes # (issue)
https://github.com/appsmithorg/appsmith/issues/22038

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Manual on storybook 
- Manual on main repo
- Jest
- Cypress

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: packages/design-system/src/SegmentedControl/SegmentedControl.types.ts [hunk 1]
```diff
@@ -22,4 +22,6 @@ export type SegmentedControlProps = {
   defaultValue?: string;
   /** The onChange of the SegmentedControl */
   onChange?: (value: string) => void;
+  /** Whether the control should consume full width of container or not */
```
Comment (tanvibhakta) [inline packages/design-system/src/SegmentedControl/SegmentedControl.types.ts:25]: Please add a TODO talking about why this is a bad idea but why it has been done anyway and what is needed to refactor this.

File: packages/design-system/src/Icon/Icon.provider.tsx [hunk 2]
```diff
@@ -951,7 +553,6 @@ const ICON_LOOKUP = {
   setting: SettingIcon,
   share: ShareForwardIcon,
   shine: ShineIcon,
-  slash: SlashIcon,
```
Comment (tanvibhakta) [inline packages/design-system/src/Icon/Icon.provider.tsx:954]: have you removed slash entirely? We need it for the API page, yes?

File: packages/design-system/src/Icon/Icon.provider.tsx [hunk 3]
```diff
@@ -986,15 +587,13 @@ export function IconProvider(props: {
...
-    .split("-")
-    .map((word) => word[0].toUpperCase() + word.slice(1))
-    .join("");
+  // const pascalCaseIconName = iconName
+  //   .split("-")
+  //   .map((word) => word[0].toUpperCase() + word.slice(1))
+  //   .join("");
 
   const Icon = ICON_LOOKUP[iconName as keyof typeof ICON_LOOKUP]
     ? ICON_LOOKUP[iconName as keyof typeof ICON_LOOKUP]
```
Comment (tanvibhakta) [inline packages/design-system/src/Icon/Icon.provider.tsx:596]: Does this mean we can use IconName as a type again? If yes, we should update all the places we are using icon - button, link, select, etc to accept an IconName and not a String.

**Review by tanvibhakta — CHANGES_REQUESTED** [review]

File: packages/design-system/src/Icon/Icon.provider.tsx (hunk 3 above)
Comment (albinAppsmith) [reply packages/design-system/src/Icon/Icon.provider.tsx:596]: We will do this after this hustle... I will add TODO here

**Review by tanvibhakta — APPROVED** [review]

**Pull request merged**